src/pd/Client.cc

```diff
@@ -184,12 +186,16 @@ void Client::initLeader()
 void Client::updateLeader()
 {
     std::unique_lock lk(leader_mutex);
-    for (const auto & url : urls)
+    for (auto it = urls.begin(); it != urls.end();)
     {
+        const auto & url = *it;
```

Comment:
```suggestion
        const auto url = *it;
```
Here we should make `url` as a copy of `*it`. Otherwise this variable is invalid in line 198